proto/accessgraph/v1alpha/access_graph_service.proto

```diff
@@ -312,3 +542,191 @@ message NetIQSyncOperation {}
+    AWSCloudTrailConfig cloud_trail_config = 1; // Effective, negotiated configuration.
+    AWSCloudTrailResumeState resume_state = 2; // Complete resume state.
+  }
+}
+
+// AWSCloudTrailResumeState holds the state for all regions.
+message AWSCloudTrailResumeState {
+  map<string, AWSCloudTrailResumeRegionState> regions_state = 1;
+  google.protobuf.Timestamp start_date = 2; // Start date for exporting CloudTrail logs.
+  google.protobuf.Timestamp end_date = 3; // End date for exporting CloudTrail logs.
+}
+
+// AWSCloudTrailResumeRegionState holds the state for a particular region.
+message AWSCloudTrailResumeRegionState {
+  string next_page = 1;
+  optional string last_event_id = 2;
+  google.protobuf.Timestamp last_event_time = 3;
+}
+
+// AWSCloudTrailEvent holds the raw unstructured log event data and its metadata.
+message AWSCloudTrailEvent {
+  // The Amazon Web Services access key ID that was used to sign the request.
+  // If the request was made with temporary security credentials, this is the
+  // access key ID of the temporary credentials.
+  string access_key_id = 1;
+
+  // A JSON that contains a representation of the event returned.
+  google.protobuf.Struct cloud_trail_event = 2;
+
+  // The CloudTrail ID of the event returned.
+  string event_id = 3;
+
+  // The name of the event returned.
+  string event_name = 4;
+
+  // The Amazon Web Services service to which the request was made.
+  string service_source = 5;
+
+  // The date and time of the event returned.
+  google.protobuf.Timestamp end_time = 6;
+
+  // Information about whether the event is a write event or a read event.
+  bool read_only = 7;
+
+  // A list of resources referenced by the event returned.
+  repeated AWSCloudTrailEventResource resources = 8;
+
+  // A user name or role name of the requester that called the API in the event
+  // returned.
+  string username = 9;
+
+  // The AWS account id of the account where the log is originated from.
+  string aws_account_id = 10;
+}
+
+// AWSCloudTrailEventResource identifies the AWS resource by name and type.
+message AWSCloudTrailEventResource {
+  // The name of the resource referenced by the event returned. These are user-created
+  // names whose values will depend on the environment. For example, the resource
+  // name might be "auto-scaling-test-group" for an Auto Scaling Group or "i-1234567"
+  // for an EC2 Instance.
+  string name = 1;
+
+  // The type of a resource referenced by the event returned. When the resource
+  // type cannot be determined, null is returned. Some examples of resource types
+  // are: Instance for EC2, Trail for CloudTrail, DBInstance for Amazon RDS, and
+  // AccessKey for IAM. To learn more about how to look up and filter events by
+  // the resource types supported for a service, see Filtering CloudTrail Events
+  // (https://docs.aws.amazon.com/awscloudtrail/latest/userguide/view-cloudtrail-events-console.html#filtering-cloudtrail-events).
+  string type = 2;
+}
```

Comment:
```suggestion
```

Reply:
I see, that must have been the thing were Roman's spending alarms went off. Interesting.